pkg/model/gcemodel/context.go

```diff
@@ -142,12 +142,12 @@ func (c *GCEModelContext) NetworkingIsGCERoutes() bool {
 
 // LinkToServiceAccount returns a link to the GCE ServiceAccount object for VMs in the given role
 func (c *GCEModelContext) LinkToServiceAccount(ig *kops.InstanceGroup) *gcetasks.ServiceAccount {
-	if c.Cluster.Spec.CloudConfig.GCEServiceAccount != "" {
+	if c.Cluster.Spec.CloudProvider.GCE.ServiceAccount != "" {
 		// This is a legacy setting because the nodes & control-plane run under the same serviceaccount
-		klog.Warningf("using legacy spec.cloudConfig.gceServiceAccount=%q setting", c.Cluster.Spec.CloudConfig.GCEServiceAccount)
+		klog.Warningf("using legacy spec.cloudProvider.gce.serviceAccount=%q setting", c.Cluster.Spec.CloudProvider.GCE.ServiceAccount)
```

Comment:
Nit: until we make v1alpha3 the default, we probably should keep the v1alpha2 names?  I recognize this is a pain though...

Reply:
We're maintaining a mapping table in cluster_spec.md. If we don't adjust as we go, we're bound to miss a few when we change the default.

Reply:
We could create a helper function that can translate between API versions (in future), even if we don't do the mapping at first...  But maybe that's good for another PR